src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -609,7 +608,7 @@ private static void collectDeclarations(Deque<AbstractFrame> frameStack, DetailA
             case TokenTypes.LITERAL_NEW:
                 if (isAnonymousClassDef(ast)) {
                     frameStack.addFirst(new AnonymousClassFrame(frame,
-                            ast.getFirstChild().toString()));
+                            ast.toString()));
```

Comment:
The token is `LITERAL_NEW`, so the mutation difference here is giving a frame the name `new` (from `LITERAL_NEW`) or giving what we wanted which was the name of the class being instantiated.

`AnonymousClassFrame` doesn't track the `AST` so it all depends on how it uses the name.

The only place it seems to be used is `logViolation`. It compares the name to the result of `getNearestClassFrameName` but it looks like it is getting the name from the same frame as defined as the "nearest".

It is comparing apples to apples, so it seems it may not care what the name is as long as it matches itself. Maybe we could confuse it with the same name multiple times, but this may require a deeper analysis.

Reply:
So I am not finding anything more to kill this. It does seem like the name doesn't matter since we are comparing the same frame to each other for anonymous classes. `getNearestClassFrameName` finds the first `CLASS_FRAME`. We can't put some other class into the anonymous class and it doesn't seem like the frame scans upper frames that much.

I don't agree that this change is valid, but I can't find anyway to kill it naturally.

Is it ok to make a reflective test to kill this?

Reply:
So, all anonymous class frames are named `new`?

Reply:
>We can't put some other class into the anonymous class

Not sure what exactly this means, can you elaborate?

Reply:
Can we just remove name field from this class if it doesn’t matter?

Reply:
We can remove the name field, but we can't remove the override for `getFrameName`. Returning `null` for it gives an NPE. Giving the AST in the constructor to the super of `AnonymousClassFrame` gives an invalid difference in the tests as it then thinks its a class instead of an anonymous at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L493 .

I am still not liking some of this code for this check.

Reply:
lets make reflection hack to kill mutation and in comment explain that we keep this junit until some issue where we will need to rethink such survivals and kill them with better design or ......
I am ok sacrifice our rule on "no pure junit", as we canont keep comments in suppression xml files and we can do this in java.